alf/bin/grid_search.py

```diff
@@ -302,11 +308,22 @@ def _worker(self, root_dir, parameters, device_queue):
             logging.set_verbosity(logging.INFO)
 
             logging.info("Search parameters %s" % parameters)
-            with gin.unlock_config():
-                gin.parse_config(
-                    ['%s=%s' % (k, v) for k, v in parameters.items()])
-                gin.parse_config(
-                    "TrainerConfig.confirm_checkpoint_upon_crash=False")
+
+            if is_gin:
+                # re-bind gin conf params
+                with gin.unlock_config():
+                    gin.parse_config(
+                        ['%s=%s' % (k, v) for k, v in parameters.items()])
+                    gin.parse_config(
+                        "TrainerConfig.confirm_checkpoint_upon_crash=False")
+            else:
+                # re-bind alf conf params
+                alf.config(parameters)
+                alf.config(
+                    'TrainerConfig', confirm_checkpoint_upon_crash=False)
```

Comment:
This won't work. If conf file is parsed (executed), changing config later is not guaranteed to have effect. Conf file has to run after these config()s (and they need to be changed to preconfig()s). One possibility is to make a copy the conf file in the main process and parse them in the worker later.

Reply:
Copying also has its own problems. Because conf is a python executable, it can import any kind files. It's hard to track those files.